src/parser.ts

```diff
@@ -0,0 +1,39 @@
+interface Token {
+  type: string;
+  value?: string;
+}
+
+function* tokenizer(input : String) : Generator<Token, Token, void> {
```

Comment:
I should get rid of the explicit type assertion, I was just trying to get things working.